### PR TITLE
Allow to remove tracks by pressing Delete key

### PR DESCRIPTION
**Pull request by ninomp**
This PR is trying to fix a highly requested feature: ability to delete/remove tracks by pressing Delete key on the keyboard.

I have pointed this to 2.3, feel free to re-point to main if you think this should not go to stable branch.

This fixes:
- https://bugs.launchpad.net/mixxx/+bug/1227676
- https://bugs.launchpad.net/mixxx/+bug/1851457
- https://bugs.launchpad.net/mixxx/+bug/1856402
- https://bugs.launchpad.net/mixxx/+bug/1944565

### CODE REVIEW

**Comment by Be-ing** [comment]
This is a new feature. It should definitely go to the main branch, not the stable branch.

**Comment by ronso0** [comment]
: )

**Comment by ninomp** [comment]
OK I will rebase.

**Review by uklotzde — CHANGES_REQUESTED** [review]
This seems to be a straightforward and easy extension. Unfortunately, we cannot add it without a command history and undo support.

Situation: You want to quickly start a new search or simply delete the existing search query. Therefore you press *Ctrl+A* followed by *Del*. Unfortunately, you didn't notice that the track table was focused instead of the search edit field. Now your carefully crafted playlist or crate is gone, forever. No chance to restore it if you don't have a database backup.

**Comment by ronso0** [comment]
I think a confirm dialog could prevent that, incl. a checkbox `[  ] Remember decision for this session".

**Comment by foss-** [comment]
Please keep in mind that for macOS laptop keyboards or compact external keyboards there is no del key. `⌘Backspace` is what Apple uses in music app as keyboard shortcut to open the delete dialog.

And yes, a confirmation dialog is a requirement for any delete operation.

**Comment by ronso0** [comment]
@uklotzde 
Do you think a warning dialog is sufficient to prevent purging playlists?
Maybe if we don't add the [  ] Remember decision for this session` checkbox?
UX would be
* select tracks
* hit `Del`
* hit `Enter`

I think requiring `Enter` is okay because it wouldn't affect the remove workflow much since `Del` and `Enter` are close on the keyboard.

Besides: how do you imagine the Undo history to be implemented?

File: src/widget/wtracktableview.cpp [hunk 1]
```diff
@@ -760,6 +760,14 @@ void WTrackTableView::keyPressEvent(QKeyEvent* event) {
         // causes a track to load since we allow in-line editing
         // of table items in general
         return;
+    } else if (event->key() == Qt::Key_Delete) {
+        QModelIndexList indices = selectionModel()->selectedRows();
+        if (indices.size() > 0) {
+            TrackModel* pTrackModel = getTrackModel();
+            if (pTrackModel) {
+                pTrackModel->removeTracks(indices);
```
Comment (ronso0) [inline src/widget/wtracktableview.cpp:768]:
```suggestion
                const auto plibraryTableModel = static_cast<LibraryTableModel*>(pTrackModel);
                if (plibraryTableModel) {
                    // Hide tracks if this is the main library table
                    // confirmation dialog
                    pTrackModel->hideTracks(indices);
                } else {
                    // Else remove the tracks from playlist/crate
                    // confirmation dialog
                    pTrackModel->removeTracks(indices);
                }
```

**Comment by ronso0** [comment]
Idea to extend this feature:
in **Tracks** `Del` would **hide** tracks (incl. the same warning dialog I mentioned above), though I'd appreciate the "Remember decision.." checkbox then because hiding tracks can be reverted.

**Comment by ronso0** [comment]
@ninomp Please take a look at a2e53733375ae3d4070071c716b9edf2407f3eeb, that's an easy way to show the shortcut in the menu to make it more discoverable.

**Comment by ronso0** [comment]
Also, if we want to avoid merge conflicts I could cherry-pick this commit into #4347 and take care of the dialogs and stuff.

**Comment by ninomp** [comment]
@ronso0 Sorry for delay. I'll try to finish this PR soon(ish), so it could be merged and then you can continue with your PR. Does this work for you?

**Comment by ronso0** [comment]
Sure, I just noticed that now that I have everything sorted out it would be super easy to incorporate your changes into my PR, as well as taking care of the dialogs.
But if you're still motivated and have time to finish it soonish I won't interfere further, except reviewing, of course ; )

**Review by ronso0 — CHANGES_REQUESTED** [review]

**Comment by uklotzde** [comment]
For safety we need at least a confirmation dialog that allows to disable subsequent confirmations for the current session.

**Comment by uklotzde** [comment]
> > For safety we need at least a confirmation dialog that allows to disable subsequent confirmations for the current session.
> 
> I think a simple confirmation dialog suffices. An option to disable subsequent confirmations may be added later on if deemed necessary.

Then a simple message box instead of a custom dialog would be sufficient. I thought about this, too. Ok, then let's start with this minimum solution.

**Review by uklotzde — CHANGES_REQUESTED** [review]

**Review by ronso0 — CHANGES_REQUESTED** [review]

**Comment by ninomp** [comment]
This is ready.

**Review by ronso0 — APPROVED** [review]

**Comment by ronso0** [comment]
Squash into one commit to keep the git history clean?

**Review by uklotzde — CHANGES_REQUESTED** [review]

**Comment by ninomp** [comment]
This is again ready.

**Review by uklotzde — APPROVED** [review]
Thank you! LGTM

**Comment by ninomp** [comment]
@foss- Could you test #4577 to check whether it solves the minor issue on macOS you reported?

**Pull request merged**